api/internal/localizer/locloader_test.go

```diff
@@ -0,0 +1,315 @@
+		"a":     outer,
+		"a/b/c": middle,
+	}
+}
+
+func TestValidateLocArgsCwdNotRoot(t *testing.T) {
+	cases := map[string]struct {
+		wd     string
+		target string
+		scope  string
+		newDir string
+	}{
+		// target not immediate child of scope
+		"outer dir": {
+			"a",
+			"b/c/d/e",
+			"b/c",
+			"b/newDir",
+		},
+		"scope": {
+			"a/b/c",
+			"d/e",
+			".",
+			"d/e/newDir",
+		},
+	}
+
+	for name, test := range cases {
+		test := test
+		t.Run(name, func(t *testing.T) {
+			req := require.New(t)
+			fSys := makeWdFs(t)[test.wd]
+
+			ll, dst, err := lclzr.ValidateLocArgs(test.target, test.scope, test.newDir, fSys)
+			req.NoError(err)
+			newDir := test.wd + "/" + test.newDir
+			checkLocLoader(req, ll, "a/b/c/d/e", "a/b/c")
+
+			req.Equal(newDir, dst.String())
+			// memory file system can only find paths rooted at current node
+			req.True(fSys.Exists(test.newDir))
+		})
+	}
+}
+
+func TestValidateLocArgsFails(t *testing.T) {
+	cases := map[string]struct {
+		target string
+		scope  string
+		dest   string
+	}{
+		"non-existent target": {
+			"/b",
+			"/",
+			"/newDir",
+		},
+		"file target": {
+			"/a/kustomization.yaml",
+			"/",
+			"/newDir",
+		},
+		"inner scope": {
+			"/alpha",
+			"/alpha/beta",
+			"/newDir",
+		},
+		"side scope": {
+			"/alpha",
+			"/a",
+			"/newDir",
+		},
+		"existing dst": {
+			"/alpha",
+			"/",
+			"/a",
+		},
+	}
+	for name, params := range cases {
+		params := params
+		t.Run(name, func(t *testing.T) {
+			var buf bytes.Buffer
+			log.SetOutput(&buf)
+			_, _, err := lclzr.ValidateLocArgs(params.target, params.scope, params.dest, makeMemoryFs(t))
+			require.Error(t, err)
+			require.Empty(t, buf.String())
+		})
+	}
+}
+
+func TestNewFails(t *testing.T) {
+	t.Run("ValidateLocArgs", func(t *testing.T) {
+		req := require.New(t)
+		fSys := makeMemoryFs(t)
+
+		ll, dst, err := lclzr.ValidateLocArgs("/alpha/beta/gamma", "alpha", "alpha/beta/gamma/newDir", fSys)
+		req.NoError(err)
+		checkLocLoader(req, ll, "/alpha/beta/gamma", "/alpha")
+		checkConfirmedDir(req, dst, "/alpha/beta/gamma/newDir", fSys)
+	})
+	cases := map[string]string{
+		"outside scope":     "../../../a",
+		"at dst":            "newDir",
+		"ancestor":          "../../beta",
+		"non-existent root": "delt",
+		"file":              "delta/kustomization.yaml",
+	}
+	for name, root := range cases {
+		root := root
+		t.Run(name, func(t *testing.T) {
+			fSys := makeMemoryFs(t)
+
+			ll, _, err := lclzr.ValidateLocArgs("/alpha/beta/gamma", "alpha", "alpha/beta/gamma/newDir", fSys)
+			require.NoError(t, err)
+
+			_, err = ll.New(root)
+			require.Error(t, err)
+		})
+	}
+}
+
+func TestLoadFails(t *testing.T) {
+	t.Run("ValidateLocArgs", func(t *testing.T) {
+		req := require.New(t)
+		fSys := makeMemoryFs(t)
+
+		ll, dst, err := lclzr.ValidateLocArgs("a", "", "/a/b/newDir", fSys)
+		req.NoError(err)
+		checkLocLoader(req, ll, "/a", "/a")
+		checkConfirmedDir(req, dst, "/a/b/newDir", fSys)
+	})
+	cases := map[string]string{
```

Comment:
Since this test is just checking for load failures, and there don't seem to any errors in lines 284-293, why are those lines there? Do they belong in another test?

Reply:
Yes, my bad 🤦‍♀️! I must've changed line 307 without changing lines 284-293. Basically, the first test is supposed to check that the factory method checks out before testing `Load`. Will fix.